iroh-bytes/src/util.rs

```diff
@@ -13,10 +13,41 @@ pub mod io;
 pub mod progress;
 pub mod runtime;
 
+/// A format identifier
+///
+/// Should we make this an u64 and use <https://github.com/multiformats/multicodec/blob/master/table.csv>?
+///
+/// That table is so weird. There is so much unrelated stuff in there, so the smallest value we would be
+/// able to use for iroh collections would be 2 bytes varint encoded or something...
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum BlobFormat {
+    /// Hash refers to a blob
+    Raw,
+    /// Hash refers to an iroh collection (format here?)
+    Collection,
+}
+
+/// A hash and format pair
+pub type Cid = (Hash, BlobFormat);
```

Comment:
can we not call these `cid`s? given that this is not the same as multiformats cids

Reply:
well, this is the question.

These are basically a blake3 cid (hash type and size is fixed, but format is variable). Depending on how we decide regarding the format, these might actually serialize as multiformats cids.

Reply:
Something like this:

```
enum BlobFormat {
  Raw,
  Collection,
  Multiformat(u64),
}

impl BlobFormat {
  fn to_multiformat(&self) -> u64 {
    ...
  }
  fn from_multiformat(value: u64) -> Self {
    ...
  }
}

struct Blake3Cid(Hash, BlobFormat);

impl Blake3Cid {
  pub fn as_cid_bytes(&self) -> Vec<u8> {
    ...
  }
  pub fn from_cid_bytes(bytes: &[u8]) -> Option<Self> {
    ...
  }
}
```

Reply:
I was planning to replace this tuple with a proper struct called `Blake3Cid` or something that has the various ser/de methods.

Reply:
made a tuple struct now. I think we can confine all cid craziness to a helper struct. The only nod to multiformats is the values we use in format.